zkvm/src/signature/signature.md

```diff
@@ -0,0 +1,44 @@
+# Signatures: Engineering design doc
+
+vanilla design:
+- single message 
+- no recursion
+- no defense against russell's attack
+
+2 parts of protocol:
+1. key aggregation
+	multikey
+	keeps transcript, feeds pubkeys to transcript
+	clone / fork transcript to make per-key factors
+2. signing & verifying
+	each party has state, wants to go to next state by verifying & performing computations. so you can "encode validity of something in the type".
+	party transitions need to be non-replayable (is this true?)
+
+dealer & parties in rangeproof:
+- dealer accumulates all factors, holds onto the transcript, gives challenges
+- parties don't have to know about other parties, who's failing, transcript, etc.
+
+oleg says: shouldn't have dealer
```

Comment:
Yeah, because each party has to validate what they sign, and each other's contributions before they proceed. In RP MPC it is not that important because each party proves their own statement and dealer only stitches those together, - parties individually are not critically affected by misbehavior of others. E.g. cannot be fooled to leak the key or accidentally prove something else ("replay a signature"). Also in RP MPC we explicitly do not have a goal of verifying transcript, so parties are simply proving the RP "contextless", that is shifted to the outer protocol, which possibly would have to replay dealer's state like we would replay counterparties' states.